lib/dialyzer/src/erl_bif_types.erl

```diff
@@ -1536,6 +1536,14 @@ type(lists, zipwith3, 4, Xs, Opaques) ->
                                         t_nil()) end, Opaques);
 
 %%-- maps ---------------------------------------------------------------------
+type(maps, from_keys, 2, Xs, Opaques) ->
+  strict(maps, from_keys, 2, Xs,
+	 fun ([List, Value]) ->
+	     case t_is_nil(List, Opaques) of
+	       true -> t_from_term(#{});
+	       false -> t_map([], t_list_elements(List, Opaques), Value)
+	     end
+	 end, Opaques);
```

Comment:
Please review this one carefully because it is the first time I change `erl_bif_types`.

Reply:
You need to update `erl_bif_types:arg_types/3` too:

```erlang
arg_types(maps, from_keys, 2) ->
  [t_list(), t_any()];
```

There's also `beam_call_types:types/3`, which the compiler uses.

```erlang
types(maps, from_keys, [Keys, Value]) ->
    RetType = #t_map{super_key=erlang_hd_type(Keys),
                     super_value=Value},
    sub_unsafe(RetType, [proper_list(), any]);
```